solarforecastarbiter/io/nwp.py

```diff
@@ -41,4 +59,93 @@ def load_forecast(latitude, longitude, init_time, start, end, model,
     ------
     ValueError : Raised if the requested variable is not found.
     """
-    raise NotImplementedError
+
+    filepath = (Path(base_path) / model /
+                init_time.strftime('%Y/%m/%d/%H') / (model + '.nc'))
+
+    mapping_subset = {k: v for k, v in CF_MAPPING.items() if v in variables}
+
+    limit = 500  # maximum distance from point to closest grid point
+
+    with xr.open_dataset(filepath) as ds:
+        # might want to check for points that are (too far) out of the domain
+        try:
+            pnt = ds.sel(latitude=latitude, longitude=longitude,
+                         method='nearest')
+        except ValueError:
+            iy_min, ix_min = tunnel_fast(
+                ds.latitude.values, ds.longitude.values,
+                latitude, longitude, limit=limit)
+            pnt = ds.sel(y=iy_min, x=ix_min, method='nearest')
+        else:
+            # we already have the point, but use tunnel_fast to determine
+            # if it's close enough
+            tunnel_fast(ds.latitude.values[:, np.newaxis],
```

Comment:
why newaxis here but not above? Do you mean to select lat/lon at the point already found? If not, why not just use tunnel_fast for both options?

Reply:
the reason was to ensure that the point was within `limit` of the nearest grid box. new code does the same thing, but hopefully more clear.